dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoriesResource.java

```diff
@@ -554,6 +562,127 @@ public final void export(@Context final HttpServletRequest httpRequest,
+    private Response processImport(final HttpServletRequest httpRequest,
+            final HttpServletResponse httpResponse,
+            final File uploadedFile,
+            final FormDataMultiPart multiPart,
+            final FormDataContentDisposition fileDetail,
+            final String filter,
+            final String exportType,
+            final String contextInode) throws IOException {
+
+        List<Category> unableToDeleteCats = null;
+        final List<FailedResultView> failedToDelete = new ArrayList<>();
+
+        BufferedReader br = null;
+        StringReader sr = null;
+
+        try {
+            final InitDataObject initData = webResource.init(null, httpRequest, httpResponse, true,
+                    null);
+
+            final User user = initData.getUser();
+            final PageMode pageMode = PageMode.get(httpRequest);
+
+            Logger.debug(this, () -> "Importing the list of categories. " + String.format(
+                    "Request payload is : {contextInode : %s, filter : %s, exportType : %s}",
+                    contextInode,
+                    filter, exportType));
+
+            String content = getContentFromFile(uploadedFile.getPath());
+
+            sr = new StringReader(content);
+            br = new BufferedReader(sr);
+
+            if (exportType.equals("replace")) {
+                Logger.debug(this, () -> "Replacing categories");
+                if (UtilMethods.isSet(contextInode)) {
+                    Category contextCat = this.categoryAPI.find(contextInode, user, false);
+                    unableToDeleteCats = this.categoryAPI.removeAllChildren(contextCat, user,
+                            false);
+                    if (!unableToDeleteCats.isEmpty()) {
+                        for (final Category category : unableToDeleteCats) {
+                            Logger.error(this, "Category with Id: " + category.getInode()
+                                    + " unable to delete");
+                            failedToDelete.add(new FailedResultView(category.getInode(),
+                                    "Category with id: " + category.getInode()
+                                            + " unable to delete"));
+                        }
+                    }
+                } else {
+                    Logger.debug(this, () -> "Deleting all the categories");
+                    categoryAPI.deleteAll(user, false);
+                    Logger.debug(this, () -> "Deleted all the categories");
+                }
+
+                this.categoryHelper.addOrUpdateCategory(user, contextInode, br, false);
+            } else if (exportType.equals("merge")) {
+                Logger.debug(this, () -> "Merging categories");
+                this.categoryHelper.addOrUpdateCategory(user, contextInode, br, true);
+            }
+
+        } catch (Exception e) {
+            Logger.error(this, "Error importing categories", e);
+        } finally {
+            sr.close();
```

Comment:
we have on dotCMS the CloseUtils, you can use such as

CloseUtils.closeQuietly(sr, br)

Also use meaningful names, such as stringReader, bufferedReader

Reply:
This change has been addressed